components/token-asset/TokenInfo.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { TickerInfo, useKasplex } from "@/hooks/useKasplex.ts";
+import { TokenMetadata } from "@/hooks/useKasFyi.ts";
+import { formatUSD } from "@/lib/utils.ts";
+import kasIcon from "@/assets/images/kas-icon.svg";
+import { twMerge } from "tailwind-merge";
+import LabelLoading from "@/components/LabelLoading.tsx";
+import { applyDecimal } from "@/lib/krc20.ts";
+
+export default function TokenInfo() {
+  const { ticker } = useParams();
+  const { fetchTokenInfo } = useKasplex();
```

Comment:
![image](https://github.com/user-attachments/assets/1863438d-c2e0-4716-a0da-36246fd6b7d2)
I would make sure `fetchTokenInfo` be called once when going into token asset page. Maybe it would be better to have parent component to handle the cache and refresh.